usr/lib/pkgconfig/sdl2.pc

```diff
@@ -0,0 +1,9 @@
+# sdl pkg-config source file
+
+Name: sdl2
+Description: Simple DirectMedia Layer is a cross-platform multimedia library designed to provide low level access to audio, keyboard, mouse, joystick, 3D hardware via OpenGL, and 2D video framebuffer.
+Version: 2.0.9
+Requires:
+Conflicts:
+Libs: ${NXDK_DIR}/lib/libSDL2.lib
```

Comment:
How/Where is i.e. SDL2TTF included?

Reply:
This is the pkg-config for SDL2. It's based on https://github.com/XboxDev/nxdk-sdl/blob/a2a2d9ca1a5a7082c830490a0b110829939de3a7/sdl2.pc.in (or similar)
I have to include it here, because when SDL2 is built using the nxdk makefile, then it won't generate or install this file.

This file is necessary because it drives `pkg-config` (which provides which arguments to add to the compiler / linker to include the library):

```
$ pkg-config sdl2 --cflags --libs
-I/usr/include/SDL2 -D_REENTRANT -pthread -lSDL2 
```
```
$ nxdk-pkg-config sdl2 --cflags --libs
-I/home/fox/Data/Projects/XboxDev/nxdk/lib/sdl/SDL2/include /home/fox/Data/Projects/XboxDev/nxdk/lib/libSDL2.lib 
```

SDL2_ttf etc. have their own pkg-config files, but I don't provide one for nxdk:

```
$ pkg-config SDL2_ttf --cflags --libs
-I/usr/include/SDL2 -D_REENTRANT -lSDL2_ttf -pthread -lSDL2 
```
```
$ nxdk-pkg-config SDL2_ttf.pc --cflags --libs
Package SDL2_ttf.pc was not found in the pkg-config search path.
Perhaps you should add the directory containing `SDL2_ttf.pc.pc'
to the PKG_CONFIG_PATH environment variable
Package 'SDL2_ttf.pc', required by 'virtual:world', not found
```

The reason that I don't provide such a file for SDL2_ttf in this PR, is that it's not an essential library for nxdk.

Therefore I think nxdk should not maintain it.

Existing nxdk code will continue to work fine, because the original nxdk makefile still includes SDL2_ttf.
However, people who want to use SDL2_ttf with this new build-system should install SDL2_ttf independently, which is as easy as doing this (which are the steps from the original SDL2_ttf README, except `./configure` is replaced by `nxdk-configure`):

```
git clone <repo>/SDL2_ttf.git
cd SDL2_ttf
nxdk-configure
make
make install
```

After that, `nxdk-pkg-config` will find it, because it's properly installed.

We could also install SDL2 this way, but it would require more work and add dependency on autoconf (or CMake) so nxdk becomes usable at all.
So, as a way of bootstrapping, we keep the broken nxdk makefile for SDL2 and add missing parts manually (= just making our own autoconf-less SDL2 build-system more complete, by adding files that would have been generated by autoconf).

SDL2 (and maybe lwip) are probably the only cases where such manual build-system integration is necessary.
All other parts in nxdk are non-conditional, essential or assumed to be "always enabled" by the build-system (which means we can add them in the compiler- or linker-wrappers instead).

That way, users don't need autoconf or CMake to use nxdk, but if they want to use such build-systems, they can use it.